serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/integration-tests-with-postgresql.adoc

```diff
@@ -1 +1,198 @@
+// Referenced documentation pages.
+:basic-integration-test-with-restassured: xref:testing-and-troubleshooting/basic-integration-tests-with-restassured.adoc
+:mocking-opnapi-services-with-wiremock: xref:testing-and-troubleshooting/mocking-opnapi-services-with-wiremock.adoc
+:mocking-http-cloudevents-with-wiremock: xref:testing-and-troubleshooting/mocking-http-cloudevents-with-wiremock.adoc
+// External pages
+:kogito_sw_examples_git_repo_url: https://github.com/kiegroup/kogito-examples.git
+:quarkus_testing_guide: https://quarkus.io/guides/getting-started-testing
+:quarkus_testing_guide_integration_test: https://quarkus.io/guides/getting-started-testing#quarkus-integration-test
+:quarkus_testing_guide_test_resource: https://quarkus.io/guides/getting-started-testing#quarkus-test-resource
+
+This document describes how to test your Serverless Workflow application integrated with PostgreSQL persistence.
+
+[[integration-test-overview]]
+== Overview of Quarkus Integration testing
+The goal is to be able to launch and test the artifact produced by the Quarkus builds and verify its interaction
+with a PostgreSQL database instance.
+
+In order to simulate that scenario and verify the service behaviour, we are using different Quarkus annotations and tools
+ that simplify this work like:
+
+* *@QuarkusIntegrationTest* will be used to launch and test the artifact produced by the Serverless Workflow Quarkus build. For more
+details see {quarkus_testing_guide_integration_test}[@QuarkusIntegrationTest] guide.
+* *@QuarkusTestResource* allows us to define test resources that need to be started before running the test. For more details see
+{quarkus_testing_guide_test_resource}[Starting services before the Quarkus application starts] guide.
+* *Kogito-quarkus-test-utils* with some Kogito predefined testcontainers as the PostgreSQL one.
+
+The testing case followed in this document is based on the `serverless-workflow-examples/serverless-workflow-newsletter-subscription` example application.
+You can access this example application in {kogito_sw_examples_git_repo_url}[Kogito Examples] GitHub repository
+
+
+[[creating-integration-test]]
+== Creating the integration test with postgreSQL
+Let's create the integration test for a specific Serverless Workflow with PostgresQL:
+
+First, we need to include the dependency to kogito-quarkus-test-utils at the project pom.xml
+--
+
+.Dependencies in `pom.xml` file
+[source,xml]
+----
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-quarkus-test-utils</artifactId>
+  <scope>test</scope>
+</dependency>
+----
+--
+This allows us to use predefined test resources such the one we need in this case for PostgreSQL: PostgreSqlQuarkusTestResource
+
+[NOTE]
+====
+This test framework allows the use of predefined test resources like: KafkaQuarkusTestResource, KeycloakQuarkusTestResource that can be used to test
+more complex scenarios
+====
+
```

Comment:
Can you please add references to these classes? A reader might feel lost entirely without a reference to these resources.

Reply:
@ricardozanini I've added the reference to  availableTestContainters in github. Is this what you mean with that reference?